File: clojure/math/pascal.clj
;; Print 10 rows of Pascal's triangle, taken from the lazy sequence each row of which maps the one before.
;; Run: clojure -M pascal.clj

(require '[clojure.string :as string])

(def rows
  (iterate (fn [row] (mapv + (cons 0 row) (conj row 0))) [1]))

(doseq [row (take 10 rows)]
  (println (string/join " " row)))

File: scala/math/Pascal.scala
// Print 10 rows of Pascal's triangle, taken from the lazy list each row of which zips the one before.
// Run: scala-cli run Pascal.scala

val rows: LazyList[Vector[Long]] =
  LazyList.iterate(Vector(1L))(row => (0L +: row).zip(row :+ 0L).map((left, right) => left + right))

@main def pascalMain(): Unit =
  rows.take(10).foreach(row => println(row.mkString(" ")))
